src/components/MAPI/apps/AppsProvider.tsx

```diff
@@ -0,0 +1,103 @@
+import produce from 'immer';
+import PropTypes from 'prop-types';
+import React, {
+  createContext,
+  useCallback,
+  useContext,
+  useMemo,
+  useReducer,
+} from 'react';
+
+export type SelectedCatalogs = Record<string, boolean>;
+
+export interface IAppsContext {
```

Comment:
Made this a little more generic, so we could add some more common app functionality later